docs-src/content/functions/test.yml

```diff
@@ -37,3 +37,43 @@ funcs:
         template: <arg>:1:3: executing "<arg>" at <fail>: error calling fail: template generation failed
         $ gomplate -i '{{ test.Fail "something is wrong!" }}'
         template: <arg>:1:7: executing "<arg>" at <test.Fail>: error calling Fail: template generation failed: something is wrong!
+  - name: test.Required
+    alias: required
+    description: |
+      Passes through the given value, if it's non-empty, non-zero, and non-`nil`.
```

Comment:
Hmm... I may want to re-think this. Helm only cares if it's an empty string or `nil`, whereas my implementation looks at _any_ zero value, including `0`, `false`, empty arrays, and empty maps. This may be somewhat less useful. And besides, if the goal is to prevent specific values (like `0`), `assert` is probably more appropriate.

Reply:
Consider this:

```
nodes: {{ getenv "NODE_COUNT" | required "Must specify NODE_COUNT; use `0` for none" }}
```

I think `0` is a non-error. But assert could also be okay, I suppose. Anyways, as far as *our* use-case goes, empty (as in not defined) is what we care about.

Reply:
@osterman yeah, undefined is unfortunately not going to be handled by `required` (
https://github.com/hairyhenderson/gomplate/issues/310#issuecomment-411835644), because gomplate uses the `missingkey=error` option (and Helm doesn't).

But I think it's probably fair to scope this down to only failing on `nil` and `""` (empty string), which is what Helm does for defined values.

Reply:
Updated to only fail on `nil` and `""`